rsconnect/actions.py

```diff
@@ -463,6 +453,19 @@ def deploy_jupyter_notebook(connect_server, file_name, extra_files, new=False, a
     return app_url, log_lines
 
 
+def fake_module_file_from_directory(directory):
+    """
+    Takes a directory and invents a properly named file that though possibly fake,
+    can be used for other name/title derivation.
+
+    :param directory: the directory to start with.
+    :return: the directory plus the (potentially) fake module file.
+    """
+    app_name = abspath(directory)
+    app_name = dirname(app_name) if app_name.endswith('/') else basename(app_name)
```

Comment:
Instead of relying on consistent use of trailing slashes, what about using `isdir`?

Reply:
Because if someone actually specifies the trailing `/`, `isdir()` would still return `True`.

Reply:
Which is correct and what you want, no?  Anyway, possibly I'm missing something, you do you.